packages/core/src/data-access/commits.ts

```diff
@@ -27,6 +34,25 @@ export async function listCommits() {
   return database.select().from(commits)
 }
 
-export async function listStagedCommits() {
-  return database.select().from(commits).where(isNull(commits.nextCommitId))
+export async function isCommitDraft(
+  { uuid }: { uuid: string },
+  tx = database,
+): Promise<TypedResult<boolean, LatitudeError>> {
+  const res = await tx
+    .select()
+    .from(commits)
+    .where(eq(commits.uuid, uuid))
+    .limit(1)
+
+  if (!res.length) return Result.error(new NotFoundError('Commit not found'))
+  const commit = res[0]!
+
+  if (commit.nextCommitId) return Result.ok(false)
+
+  const commitsReferencingCommit = await tx
+    .select({ id: commits.id })
+    .from(commits)
+    .where(eq(commits.nextCommitId, commit.id))
+
+  return Result.ok(commitsReferencingCommit.length === 0)
 }
```

Comment:
data access should not answer booleans or perform business logic they just an interface to the database. Think of them as repositories.